### PR TITLE
feat: Add TOC filtering

### PR DESCRIPTION
**Pull request by BenjaminMichaelis**
Fixes #445

### CODE REVIEW

**Comment by BenjaminMichaelis** [comment]
If you have extra time to review @ascott18, always appreciate your feedback on front end things like this :) But no worries if you don't have time

File: EssentialCSharp.Web/Views/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -182,20 +181,23 @@
...
-                                <div>
-                                    <h5>
+                                <div style="display: flex; align-items: center;">
+                                    <h5 style="margin-right: 8px;">
                                         Contents
                                     </h5>
+                                    <i class="fa-solid fa-filter" v-on:click="enableTocFilter = enableTocFilter === 'filter' ? 'none' : 'filter';"></i>
+                                </div>
+                                <div v-if="enableTocFilter === 'filter'" class="filter-input-container">
+                                    <input type="text" class="filter-input" v-model="searchQuery" placeholder="Filter TOC..." />
```
Comment (Keboo) [inline EssentialCSharp.Web/Views/Shared/_Layout.cshtml:191]:
```suggestion
                                    <input type="text" class="filter-input" v-model="searchQuery" placeholder="Search sections..." />
```

**Review by Keboo — APPROVED** [review]

File: EssentialCSharp.Web/wwwroot/js/site.js [hunk 2]
```diff
@@ -78,194 +78,226 @@ const completedFeaturesList = [
...
-        sidebarShown.value = false;
-      }
-      // when making screen bigger reveal sidebar
-      else {
-        if (!sidebarShown.value) {
-          sidebarShown.value = true;
+        const enableTocFilter = ref('none');
+        const searchQuery = ref('');
+
+        const filteredTocData = computed(() => {
```
Comment (ascott18) [inline EssentialCSharp.Web/wwwroot/js/site.js:240]: Performing stateful mutations in a computed is unusual and leads to confusing spaghetti. Use a `watchEffect` instead, or remove mutations of `expandedTocs`.

Comment (BenjaminMichaelis) [reply EssentialCSharp.Web/wwwroot/js/site.js:241]: I think I have now resolved that if this looks correct? https://github.com/IntelliTect/EssentialCSharp.Web/pull/585/commits/496159828c4f957bdc4fd8b67e17384d559015a2

Comment (ascott18) [reply EssentialCSharp.Web/wwwroot/js/site.js:241]: There's still a stateful mutation in a computed on line 243.

I probably should have actually advised that you use a simple `watch` against `searchQuery`.

Comment (BenjaminMichaelis) [reply EssentialCSharp.Web/wwwroot/js/site.js:240]: Thanks for catching my obvious oversight :) Here is another crack at it: https://github.com/IntelliTect/EssentialCSharp.Web/pull/585/commits/f475110b8c637e14101f73c3aa207f32f0053013

**Review by ascott18 — APPROVED** [review]

**Pull request merged**